scripts/release_info.py

```diff
@@ -292,6 +310,188 @@ def _check_storage_models(current_release):
     return [item for item in diff_list if item.startswith('core/storage')]
 
 
+def get_blocking_bug_issue_count(repo):
+    """Returns the number of unresolved blocking bugs.
+
+    Args:
+        repo: github.Repository.Repository. The PyGithub object for the repo.
+
+    Returns:
+        int: Number of unresolved blocking bugs.
+    """
+    blocking_bugs_milestone = repo.get_milestone(number=39)
+    return blocking_bugs_milestone.open_issues
+
+
+def check_prs_for_current_release_are_released(repo):
+    """Checks that all pull requests for current release have a
+    'PR: released' label.
+
+    Args:
+        repo: github.Repository.Repository. The PyGithub object for the repo.
+
+    Returns:
+        bool. Whether all pull requests for current release have a
+            PR: released label.
+    """
+    all_prs = repo.get_pulls(state='all')
+    for pr in all_prs:
+        label_names = [label.name for label in pr.labels]
+        if 'PR: released' not in label_names and (
+                'PR: for current release' in label_names):
+            return False
+    return True
+
+
+def update_sorted_file(filepath, new_list, start_index_line):
+    """Updates the files AUTHORS and CONTRIBUTORS with a sorted list of
+    new authors or contributors.
+
+    Args:
+        filepath: str. The path of the file to update.
+        new_list: list(str). The list of new authors or contributors to
+            add to the file.
+        start_index_line: str. The line whose index determines the start
+            of authors/contributors list in the file.
+    """
+    content = []
+    with python_utils.open_file(filepath, 'r') as f:
+        content = f.readlines()
+    start_index = content.index(start_index_line) + 2
+    updated_list = new_list + content[start_index:]
+    updated_list = sorted(updated_list, key=lambda s: s.lower())
+    content = content[:start_index] + updated_list
+    with python_utils.open_file(filepath, 'w') as f:
+        for line in content:
+            f.write(line)
+
+
+def update_changelog(release_summary_content, current_release_version):
+    """Updates CHANGELOG file.
+
+    Args:
+        release_summary_content: list(str). List of lines in
+            ../release_summary.md.
+        current_release_version: str. The version of current release.
+    """
+
+    current_date = datetime.date.today().strftime('%d %b %Y')
+    python_utils.PRINT('Updating Changelog...')
+    start_index = release_summary_content.index('### Changelog:\n') + 1
+    end_index = release_summary_content.index('### Commit History:\n')
```

Comment:
This seems dependent on release_summary_content not having a section between Changelog and Commit History, which may not be preserved in the future. You might want to implement a backend test that asserts this condition is always satisfied.

Reply:
Added a test in the script itself.